src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1087,6 +1091,101 @@ def aks_get_credentials(cmd, client, resource_group_name, name, admin=False,
 }
 
 
+# pylint: disable=line-too-long
+def aks_diag(cmd, client, resource_group_name, name, storage_account_id=None):
+    mc = client.get(resource_group_name, name)
+
+    yaml_files = urlopen("https://raw.githubusercontent.com/Azure/aks-diagnostic-tool/master/deployment/aks-diagnostic.yaml")
+    after_split = yaml_files.read().decode().split("---")
+    ds_yaml = after_split[0]
+    secret_yaml = after_split[1]
+    containerlogs_cm = after_split[2]
+    kubeobjects_cm = after_split[3]
+
+    if storage_account_id is None:
+        print("No storage account specified. Try getting storage account from diag settings")
+        storage_account_id = get_storage_account_from_diag_settings(cmd.cli_ctx, resource_group_name, name)
+        if storage_account_id is None:
+            raise CLIError("Must specify a stroage account if no one is setup in diag settings.")
+
+    print("Will use storage account %s to store diag info." % storage_account_id)
+
+    from msrestazure.tools import is_valid_resource_id, parse_resource_id
+    if is_valid_resource_id(storage_account_id):
+        try:
+            parsed_storage_account = parse_resource_id(storage_account_id)
+        except CloudError as ex:
+            raise CLIError(ex.message)
+    else:
+        raise CLIError("Invalid storage account id %s" % storage_account_id)
+
+    storage_account_name = parsed_storage_account['name']
+
+    storage_client = cf_storage(cmd.cli_ctx, parsed_storage_account['subscription'])
+    storage_account_keys = storage_client.storage_accounts.list_keys(parsed_storage_account['resource_group'], storage_account_name)
+    kwargs = {
+        'account_name': storage_account_name,
+        'account_key': storage_account_keys.keys[0].value
+    }
+
+    cloud_storage_client = cloud_storage_account_service_factory(cmd.cli_ctx, kwargs)
+    sas_token = cloud_storage_client.generate_shared_access_signature(
+        'b',
+        'sco',
+        'rwdlacup',
+        datetime.datetime.utcnow() + datetime.timedelta(days=1))
+
+    from knack.prompting import prompt_y_n
+    msg = 'This will deploy a daemon set to your cluster to collect logs and diag information and save them to the storage account. Are you sure?'
```

Comment:
Maybe this should say, "This will deploy a daemon set to your cluster to collect logs and diagnostic information and save them to the specified storage account as outlined in aka.ms/AKSPeriscope. If you share access to that storage account to Azure support, you consent to the terms outlined in aka.ms/DiagConsent. Do you confirm?"